src/utilities.ts

```diff
@@ -80,39 +84,43 @@ export function extractErrorContext(object) {
 	return result
 }
 
-function _matchClosestHoF(difference) {
-	return (values, value) => {
+function _matchClosestHoF(difference: (a: number, b: number) => number) {
+	return matcher
+
+	function matcher(values: ReadonlyArray<number>, value: number): number
+	function matcher<T>(values: Record<number, T>, value: number): T
+	function matcher (values: ReadonlyArray<number>|Record<number, any>, value: any) {
 		const isArray = Array.isArray(values)
 		const isObject = typeof values === typeof {}
 
 		if (!isArray && !isObject) {
 			return
 		}
 
-		const workingValues = isArray ?
-			values :
+		const workingValues: ReadonlyArray<number> = isArray ?
+			values as ReadonlyArray<number> :
 			isObject ?
-				Object.keys(values) :
+				Object.keys(values) as any as ReadonlyArray<number> :
 				[]
 
-		let closestIndex
-		let closest
+		let closestIndex: number|undefined
+		let closest: number|undefined
 
-		workingValues
-			.map(v => difference(v, value))
+		;(workingValues as ReadonlyArray<string|number>)
```

Comment:
I'm guessing this is required due to (a) the `as` necessitating the brackets, and (b) the brackets necessitating the `;` to prevent being read as a function call?

Is there any other way to handle this? Because my eyes are burning.